smolyay/samples.py

```diff
@@ -0,0 +1,415 @@
+import abc
+import numpy
+
+
+class UnidimensionalPointSet(abc.ABC):
+    """Set of unidimensional points
+
+    A set of unique unidimensional points within a domain.
+    """
+
+    def __init__(self, domain):
+        self._points = None
+        self._domain = None
+
+        self.domain = numpy.array(domain, dtype=float)
+        self._valid_cache = False
```

Comment:
I would group valid cache with the other internal variables, and then the domain setter should take care of the casting and copying now.
```suggestion
        self._valid_cache = False

        self.domain = domain
```